### ISSUE TITLE
Support for auto key rotation in azurerm_disk_encryption_set

### ISSUE THREAD

**Issue by AliMBajwa** [hashicorp/terraform-provider-azurerm#12531]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It's possible to create/update a disk encryption set with in the AZ CLI / Portal to automatically rotate keys. There isn't an option to do this via the terraform resource.

Note: I've already tried passing a versionless key ID and i get the following error:

```
Error: parsing "https://myExampleVault.vault.azure.net/keys/key-myExampleKey": expected a versioned ID but no version in " 
   with azurerm_disk_encryption_set.example_vm,
   on disk-encryption.tf line 5, in resource "azurerm_disk_encryption_set" "example_vm":
    5:   key_vault_key_id    = data.azurerm_key_vault_key.example_vm.versionless_id
```

### New or Affected Resource(s)

* azurerm_disk_encryption_set
* data.azurerm_disk_encryption_set

### Potential Terraform Configuration

```hcl
resource "azurerm_disk_encryption_set" "example_vm" {
  name                = "disk-es-${var.virtual_machine_name}"
  resource_group_name = var.resource_group_name
  location            = var.location
  key_vault_key_id    = data.azurerm_key_vault_key.example_vm
  auto_key_rotation = true

  identity {
    type = "SystemAssigned"
  }
}
```

### References

* #0000

**Comment by AliMBajwa** [comment]
Accidentally closed this - reopened immediately after. Sorry!

**Comment by johnfriedrich** [comment]
Looks this was added in https://github.com/hashicorp/terraform-provider-azurerm/pull/13747

**Comment by fcatacut** [comment]
> Looks this was added in #13747

Correct, this was released in [v2.82.0](https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v2.82.0). This issue can be closed.

**Comment by AliMBajwa** [comment]
Thanks for letting me know on this thread. 

For anyone who reads this: I've not confirmed myself it's been added and works (since I don't work on that particular project anymore where it was needed) I'm going off of what the 2 users above have said.
